distributed/scheduler.py

```diff
@@ -2220,18 +2217,17 @@ def decide_worker_non_rootish(self, ts: TaskState) -> WorkerState | None:
             wp_vals = cast("Sequence[WorkerState]", worker_pool.values())
             n_workers: int = len(wp_vals)
             if n_workers < 20:  # smart but linear in small case
-                ws = min(wp_vals, key=operator.attrgetter("occupancy"))
+                ws = min(wp_vals, key=self.worker_objective_no_deps)
                 assert ws
-                if ws.occupancy == 0:
+                if sum(self.worker_objective_no_deps(ws)) == 0:
```

Comment:
this is weird. I don't think that the return value of an objective function should necessarily be iterable. This is a bit of a weird API, isn't it? 

I think being explicit here makes the code also much better readable. In essence, this is `ws.occupancy == 0 and ws.nbytes == 0` isn't it? tieing this to the objective function is very abstract

Reply:
The other objective function also returns a tuple, and we want a tuple for tie-breaking. Otherwise, we'd have to scale occupancy and memory into the same total "cost" somehow.

> In essence, this is `ws.occupancy == 0 and ws.nbytes == 0` isn't it? tieing this to the objective function is very abstract

Yup, it could be written this way. I just wanted a single point of truth—if the objective function changed slightly in the future, you'd have to know to update this branch. (Hopefully this branch will be removed before the objective function changes, but still.)